From 2896b697955b8985e6590c571675c5cd4a2cccca Mon Sep 17 00:00:00 2001
From: Grzegorz Jaszczyk <jaz@semihalf.com>
Date: Thu, 12 Jan 2017 16:02:18 +0100
Subject: [PATCH 23/45] usb: gadget: u3d: enable probe deferral for gpio

Since the vbus-gpio can be a gpio from pca953x io-expander, it can be not
available during usb gadget probe, therefor enable probe deferral.

Change-Id: I4fc13bc5a3b672b0e72644a142461064d7360c7c
Signed-off-by: Grzegorz Jaszczyk <jaz@semihalf.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/38534
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Omri Itach <omrii@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
---
 drivers/usb/gadget/udc/mvebu_u3d.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/usb/gadget/udc/mvebu_u3d.c b/drivers/usb/gadget/udc/mvebu_u3d.c
index bc4fda7c564c..d6bc8a069a96 100644
--- a/drivers/usb/gadget/udc/mvebu_u3d.c
+++ b/drivers/usb/gadget/udc/mvebu_u3d.c
@@ -2342,6 +2342,11 @@ static int mvc2_probe(struct platform_device *pdev)
 
 	/* setup vbus gpio */
 	cp->vbus_pin = of_get_named_gpio(pdev->dev.of_node, "vbus-gpio", 0);
+	if ((cp->vbus_pin == -ENODEV) || (cp->vbus_pin == -EPROBE_DEFER)) {
+		ret = -EPROBE_DEFER;
+		goto err_clk;
+	}
+
 	if (cp->vbus_pin < 0)
 		cp->vbus_pin = -ENODEV;
 
-- 
2.35.3

